### PR TITLE
Eng 8058 implement bin function

### PR DESCRIPTION
**Pull request by BillWhite**
Implementation of the SQL function bin.

### CODE REVIEW

**Comment by paulmartel** [comment]
End of review round.

File: tests/ee/expressions/function_test.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                                                      0,
+                                                      (DRTupleStream *)0,
+                                                      (DRTupleStream *)0)) {}
+        ~FunctionTest() {
+                delete m_pool;
+                delete m_executorContext;
+        }
+        Pool            *m_pool;
+        ExecutorContext *m_executorContext;
+};
```
Comment (paulmartel) [inline tests/ee/expressions/function_test.cpp]: Minor: Declaring these members as objects vs. object pointers would not require the explicit destructor/deletes and would slightly shrink the initializer code

File: tests/frontend/org/voltdb/regressionsuites/TestFunctionsForVoltDBSuite.java [hunk 2]
```diff
@@ -1797,6 +1797,46 @@ public void testHex() throws NoConnectionsException, IOException, ProcCallExcept
...
+        int i = 0;
+        for (long val : binInterestingValues) {
+            client.callProcedure("@AdHoc", "insert into R3(id, big) values (?, ?)", i, val);
+            ++i;
+        }
+
+        for (long val : binInterestingValues) {
+            result = client.callProcedure("@AdHoc",
+                    "select bin(big) from R3 where big = " + val).getResults()[0];
+            String binString = Long.toBinaryString(val);
```
Comment (paulmartel) [inline tests/frontend/org/voltdb/regressionsuites/TestFunctionsForVoltDBSuite.java]: Minor: only need one copy of new String[]{ Long.toBinaryString(val) }

**Comment by paulmartel** [comment]
End of review round, this time for sure. I finished pushing the Comment buttons and everything this time.

File: tests/frontend/org/voltdb/regressionsuites/TestFunctionsForVoltDBSuite.java (hunk 2 above)
Comment (BillWhite) [inline tests/frontend/org/voltdb/regressionsuites/TestFunctionsForVoltDBSuite.java]: I don't think I understand this. I don't see any arrays of strings formed by Long.toBinaryString.  I just see  several individual strings, but they are only created when needed and then destroyed.
Is this a java garbage collection issue?

**Comment by paulmartel** [comment]
Looks good.

**Pull request merged**